distributed/protocol/serialize.py

```diff
@@ -16,11 +16,19 @@
 from .utils import unpack_frames, pack_frames_prelude, frame_split_size
 
 
-serializers = {}
-deserializers = {None: lambda header, frames: pickle.loads(b''.join(frames))}
+class_serializers = {}
+class_deserializers = {}
 
 lazy_registrations = {}
 
+# generalist de-serialization for any object. The keys are the options
+# available for a comm channel or Client
+deser = {
+ 'dask': None,  # this one is special, attempts special deser first
```

Comment:
I have not implemented the "namespacing" of serlialiser names yet, so for the class-based methods, the handling is necessarily specialised for the moment. I mean, I can easily make 
```
  'dask': (dask_dumps, )
```
but not know what to put in the second item, as it would never be called as things stand with the header description in use.

Reply:
Perhaps all of the serializers should produce both a header and frames and all of the deserializers should consume header and frames.  Then it would be straightforward for the dask family of serializers to operate as-is as a single serializer.

Reply:
In the case of pickle and msgpack these would just return `{}, pickle.dumps(foo)` or something